frontend init-containers: Includes init containers to pod containers

vyncent-t opened the pull request:
For issue #983 

Creates an additional field for Init containers under the regular containers when in the pod view.

<img width="695" alt="image" src="https://user-images.githubusercontent.com/78232183/224773810-dcc3e24a-fc63-4589-b533-e0a006664ded.png">

joaquimrocha commented on frontend/src/components/common/Resource/Resource.tsx:
```diff
@@ -739,6 +739,20 @@ export function ContainersSection(props: { resource: KubeObjectInterface | null
   }
 
+  function getInitContainers() {
+    if (!resource) {
+      return [];
+    }
+
+    let initContainers: KubeContainer[] = [];
+
+    if (resource.spec.initContainers) {
+      initContainers = resource.spec.initContainers;
+    }
+
+    return initContainers;
```
I think this can be simplified with `return resource?.spec?.initContainers || [];`

joaquimrocha commented on frontend/src/components/common/Resource/Resource.tsx:
```diff
@@ -780,6 +795,25 @@ export function ContainersSection(props: { resource: KubeObjectInterface | null
+        </SectionBox>
+      ) : (
+        <>
+          <SectionBox title={t(`resource| Init Containers`)} />
```
The namespace separator (`|`) shouldn't have spaces around it.

joaquimrocha commented on frontend/src/components/common/Resource/Resource.tsx:
```diff
@@ -780,6 +795,25 @@ export function ContainersSection(props: { resource: KubeObjectInterface | null
+      ) : (
+        <>
+          <SectionBox title={t(`resource| Init Containers`)} />
+          {initContainers.map((initContainer: any, i: number) => (
+            <SectionBox key={i} outterBoxProps={{ pt: 1 }}>
```
I know you just used what's written for the regular containers, but the `key` should be textual, so `key={`init_container_${i}`}` is better; and `initContainer` should have the proper type (`KubeContainer` I think).

joaquimrocha replied:
Feel free to also modify the regular container's key+type in a different commit if you want.

joaquimrocha reviewed (changes requested):
Left some comments in the code.
Also, the commit title shouldn't have `init containers` in the prefix, just frontend is enough.

If you want to reuse the exact same code, you could also pass an "isInit" property to the `ContainersSection`, and then add a second `<ContainerSection isInit={true}...`.

vyncent-t commented:
[x] updated the build snapshots
[x] commit title edit
[x] simplified with return resource?.spec?.initContainers || []
[x] namespace separator (|) shouldn't have spaces around it
[x] contextual edits

- left the original context container as is, will change with another commit as mentioned

joaquimrocha commented on frontend/src/components/common/Resource/Resource.tsx:
```diff
@@ -780,6 +785,25 @@ export function ContainersSection(props: { resource: KubeObjectInterface | null
       </>
+
+      {initContainers.length === 0 ? (
+        <SectionBox>
+          <Empty>{t('resource|No data to be shown.')}</Empty>
+        </SectionBox>
```
We should not display anything if there are no initContainers.

joaquimrocha commented on frontend/src/components/common/Resource/Resource.tsx:
```diff
@@ -780,6 +785,25 @@ export function ContainersSection(props: { resource: KubeObjectInterface | null
+        </SectionBox>
+      ) : (
+        <>
+          <SectionBox title={t(`resource|Init Containers`)} />
```
Please use single quotes here.
